pkg/endpoint/endpoint.go

```diff
@@ -290,6 +291,15 @@ type Endpoint struct {
 	// All fields within the PolicyKey and the proxy port must be in host byte-order.
 	desiredMapState PolicyMapState
 
+	// BPFConfigMap provides access to the endpoint's BPF configuration.
+	bpfConfigMap *bpfconfig.EndpointConfigMap
```

Comment:
Maybe we should start thinking on having a `desiredState` structure and a `realizedState` structure that contains all of these maps inside of that structure.

Reply:
Agreed.

Reply:
FYI I am working on something like this that is coming soon in an upcoming PR.